inc/class-storefront.php

```diff
@@ -336,12 +336,33 @@ public function get_structured_data() {
 			} else {
 				$structured_data = $structured_data + self::$structured_data[0];
 			}
-      array_walk_recursive( $structured_data, function ( &$value ) {
-        $value = sanitize_text_field( $value );
-      } );
+
+			$structured_data = $this->sanitize_structured_data( $structured_data );
 
 			echo '<script type="application/ld+json">' . wp_json_encode( $structured_data ) . '</script>';
 		}
+
+		/**
+		 * Sanitize structured data.
+		 *
+		 * @param  array $data
+		 * @return array
+		 */
+		public function sanitize_structured_data( $data ) {
+			$sanitized = array();
+
+			foreach ( $data as $key => $value ) {
+				if ( is_array( $value ) ) {
+					$sanitized_value = $this->sanitize_structured_data( $value );
+				} else {
+					$sanitized_value = sanitize_text_field( $value );
+				}
+
+				$sanitized[ sanitize_text_field( $key ) ] = $sanitized_value;
+			}
+
+			return $sanitized;
+		}
```

Comment:
@claudiosmweb 
I forgot that I had to sanitize keys...
Also, you just learn me something pretty usefull with the method/function calling itself which by this way, recursively walk through the array, thanks...